### PR TITLE
Pick docker image python version from environments python.

### PR DESCRIPTION
**Pull request by akruszewski**
## Motivation and Context
Why was this PR created?

## How has this been tested?
What testing strategies have you used?

## Checklist

- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Assigned myself to the PR
- [ ] Added tests to cover my changes
- [ ] Added new entries to the `RELEASE.md` file

## Notice

- [ ] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -33,7 +33,9 @@
 
 from setuptools import setup
 
-name = "kedro-docker"
+from kedro_docker.package_info import name
+
+# name = "kedro-docker"
```
Comment (akruszewski) [inline setup.py]: remove this line

File: kedro_docker/plugin.py [hunk 2]
```diff
@@ -196,6 +223,15 @@ def _mount_info() -> Dict[str, Union[str, Tuple]]:
     return res
 
 
+def _is_python_version_supported(python_version: str) -> bool:
```
Comment (akruszewski) [inline kedro_docker/plugin.py]: made it pure (take also python_requires as argument)

File: kedro_docker/plugin.py [hunk 3]
```diff
@@ -173,11 +177,34 @@ def docker_build(uid, gid, spark, image, docker_args):
...
+        ("--build-arg", "KEDRO_UID={0}".format(uid)),
+        ("--build-arg", "KEDRO_GID={0}".format(gid)),
+    ]
+
+    if _is_python_version_supported(
+        "{major}.{minor}.{micro}".format(
+            major=sys.version_info.major,
+            minor=sys.version_info.minor,
+            micro=sys.version_info.micro,
+        )
```
Comment (turn1a) [inline kedro_docker/plugin.py]:
```suggestion
        f"{sys.version_info.major}.{sys.version_info.minor}.{sys.version_info.micro}"
```

File: kedro_docker/plugin.py [hunk 4]
```diff
@@ -196,6 +223,15 @@ def _mount_info() -> Dict[str, Union[str, Tuple]]:
     return res
 
 
+def _is_python_version_supported(python_version: str) -> bool:
+    python_requires = distribution(name,).metadata["Requires-Python"].split(",")
```
Comment (akruszewski) [inline kedro_docker/plugin.py]: remove coma in distribution arg list

File: kedro_docker/plugin.py [hunk 5]
```diff
@@ -173,11 +177,34 @@ def docker_build(uid, gid, spark, image, docker_args):
...
+            minor=sys.version_info.minor,
+            micro=sys.version_info.micro,
+        )
+    ):
+        req_args.append(
+            (
+                "--build-arg",
+                "IMAGE_VERSION=python:{major}.{minor}-stretch".format(
+                    major=sys.version_info.major, minor=sys.version_info.minor,
+                ),
```
Comment (turn1a) [inline kedro_docker/plugin.py:198]:
```suggestion
                f"IMAGE_VERSION=python:{sys.version_info.major}.{sys.version_info.minor}-stretch",
```

File: kedro_docker/plugin.py [hunk 6]
```diff
@@ -173,11 +177,34 @@ def docker_build(uid, gid, spark, image, docker_args):
...
+            )
+        )
+    else:
+        click.echo(
+            "Python version obtained from env ({0}) is not supported.Using "
+            "default (3.6)".format(sys.version)
```
Comment (turn1a) [inline kedro_docker/plugin.py:204]:
```suggestion
            f"Python version obtained from env ({sys.version}) is not supported. Using "
            "default (3.6)"
```

File: kedro_docker/plugin.py (hunk 4 above)
Comment (turn1a) [inline kedro_docker/plugin.py]:
```suggestion
    python_requires = distribution(name).metadata["Requires-Python"].split(",")
```

File: kedro_docker/plugin.py (hunk 2 above)
Comment (turn1a) [inline kedro_docker/plugin.py]: Missing docstring.

File: setup.py (hunk 1 above)
Comment (turn1a) [inline setup.py]: Remove commented code.

File: kedro_docker/package_info.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+# license to the QuantumBlack Trademarks. You may not use the QuantumBlack
+# Trademarks or any confusingly similar mark as a trademark for your product,
+#     or use the QuantumBlack Trademarks in any other manner that might cause
+# confusion in the marketplace, including but not limited to in advertising,
+# on websites, or on software.
+#
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+""" Package metadata. """
```
Comment (turn1a) [inline kedro_docker/package_info.py:29]:
```suggestion
"""Package metadata."""
```

**Review by turn1a — CHANGES_REQUESTED** [review]
Looks good, only minor comments.

Not sure how this Python version checking in `_is_python_version_supported ` is working exactly. Could you include some unit tests?

**Pull request closed without merging**